catalyst-gateway/bin/src/service/api/documents/post_document_index_query/mod.rs

```diff
@@ -116,6 +122,20 @@ async fn fetch_docs(
         )
         .await?;
 
+    // need `FullSignedDoc` in order to read the `raw` part to check its deprecation.
+    // this is performance hit, but we use this endpoint for transition temporarily.
+    let tasks = indexed_docs.iter().flat_map(|(id, docs)| {
+        docs.iter()
+            .map(|item| async move { FullSignedDoc::retrieve(id, Some(item.ver())).await })
+            .collect::<Vec<_>>()
+    });
+
+    let results = join_all(tasks).await;
```

Comment:
Filter out during this step all the **non** deprecated documents so during the `// convert to output response` you will have all expected `deprecated` documents only